airflow/www/views.py

```diff
@@ -3012,6 +3015,56 @@ def action_muldelete(self, items):
         self.update_redirect()
         return redirect(self.get_redirect())
 
+    @action(
+        'mulduplicate',
+        'Duplicate',
+        'Are you sure you want to duplicate the selected connections?',
+        single=False,
+    )
+    @provide_session
+    @auth.has_access(
+        [
+            (permissions.ACTION_CAN_CREATE, permissions.RESOURCE_CONNECTION),
+        ]
+    )
+    def action_mulduplicate(self, connections, session=None):
+        """Duplicate Multiple connections"""
+        for selected_conn in connections:
+            new_conn_id = selected_conn.conn_id
+            matches = re.search(r"_copy(\d+)$", selected_conn.conn_id)
+            if matches:
+                new_conn_id = selected_conn.conn_id.replace(
+                    f'_copy{matches.group(1)}', f'_copy{int(matches.group(1)) + 1}'
+                )
```

Comment:
```suggestion
                conn_id_prefix = selected_conn.conn_id[:matches.start()]
                new_conn_id = f"{conn_id_prefix}_copy{int(matches.group(1)) + 1}"
```

The current implementation would fail in edge cases like `conn_id_copy1_copy1` (incorrectly replacing the first `copy1`).

Also the variable should probably be named `match` (singular) instead.

Reply:
@uranusjr, 
it won't fail in any cases as we are only matching the last occurence in the regex.
**r"_copy(\d+)$"**

Reply:
I agree,
that the name of variable should be singular.